src/library/scala/collection/immutable/Vector.scala

```diff
@@ -310,145 +353,158 @@ final class Vector[+A] private[immutable] (private[collection] val startIndex: I
   }
 
   override def prepended[B >: A](value: B): Vector[B] = {
-    val result = if (endIndex != startIndex) {
-      val blockIndex = (startIndex - 1) & ~31
-      val lo = (startIndex - 1) & 31
-
-      if (startIndex != blockIndex + 32) {
-        val s = new Vector(startIndex - 1, endIndex, blockIndex)
-        s.initFrom(this)
-        s.dirty = dirty
-        s.gotoPosWritable(focus, blockIndex, focus ^ blockIndex)
-        s.display0(lo) = value.asInstanceOf[AnyRef]
+    val thisLength = length
+    val result =
+      if (depth == 1 && thisLength < 32) {
```

Comment:
This branch and the Vector.single branch are the only changed branches, the reset of this method is just indentation